processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -109,6 +118,27 @@ public SearchQueryRunner(Segment segment)
       final ImmutableBitmap baseFilter =
           filter == null ? null : filter.getBitmapIndex(new ColumnSelectorBitmapIndexSelector(bitmapFactory, index));
 
+      ImmutableBitmap timeFilteredBitmap;
+      if (!interval.contains(segment.getDataInterval()))
+      {
+        MutableBitmap timeBitmap = bitmapFactory.makeEmptyMutableBitmap();
+        final Column timeColumn = index.getColumn(Column.TIME_COLUMN_NAME);
+        final GenericColumn timeValues = timeColumn.getGenericColumn();
+
+        int startIndex = Math.max(0, getStartIndexOfTime(timeValues, interval.getStartMillis(), true));
+        int endIndex = Math.min(timeValues.length() - 1, getStartIndexOfTime(timeValues, interval.getEndMillis(), false));
+
+        for (int i = startIndex; i <= endIndex; i++) {
+          timeBitmap.add(i);
+        }
+
+        final ImmutableBitmap finalTimeBitmap = bitmapFactory.makeImmutableBitmap(timeBitmap);
+        timeFilteredBitmap =
+            (baseFilter == null) ? finalTimeBitmap : finalTimeBitmap.intersection(baseFilter);
```

Comment:
Just ran SearchBenchmark on this PR, once as-is, and the other by setting:

```
    // Closing this will cause segfaults in unit tests.
    //final QueryableIndex index = segment.asQueryableIndex();
    final QueryableIndex index = null;
```

in SearchQueryRunner. Looks like the time bitmap is significantly faster than the Cursor scan.

TIme bitmap:
```
Benchmark                                    (limit)  (numSegments)  (rowsPerSegment)  (schemaAndQuery)  Mode  Cnt       Score      Error  Units
SearchBenchmark.queryMultiQueryableIndex        1000              1            750000           basic.A  avgt   25   19445.166 ±  570.923  us/op
SearchBenchmark.querySingleIncrementalIndex     1000              1            750000           basic.A  avgt   25  311990.777 ± 9461.168  us/op
SearchBenchmark.querySingleQueryableIndex       1000              1            750000           basic.A  avgt   25   20158.354 ±  383.347  us/op
```

Cursor scan:
```
Benchmark                                    (limit)  (numSegments)  (rowsPerSegment)  (schemaAndQuery)  Mode  Cnt       Score      Error  Units
SearchBenchmark.queryMultiQueryableIndex        1000              1            750000           basic.A  avgt   25  437587.995 ± 5194.361  us/op
SearchBenchmark.querySingleIncrementalIndex     1000              1            750000           basic.A  avgt   25  307165.009 ± 4843.741  us/op
SearchBenchmark.querySingleQueryableIndex       1000              1            750000           basic.A  avgt   25  464304.740 ± 8042.318  us/op
```